CHANGELOG.md

```diff
@@ -3,6 +3,31 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) (+ the Migration Guide),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+### Deprecated
+
+- The `create_group_labels` method of the `Dataset` class is deprecated and will be removed in the next major release.
```

Comment:
major release -> future release... I am a little inconsistent with my semver usage. Don't want to give false promises :D